xarray/core/missing.py

```diff
@@ -619,6 +619,19 @@ def interp(var, indexes_coords, method, **kwargs):
     # default behavior
     kwargs["bounds_error"] = kwargs.get("bounds_error", False)
 
+    # check if the interpolation can be done in orthogonal manner
+    if (
+        len(indexes_coords) > 1
+        and method in ["linear", "nearest"]
+        and all(dest[1].ndim == 1 for dest in indexes_coords.values())
+        and len(set([d[1].dims[0] for d in indexes_coords.values()]))
```

Comment:
This condition is confusing me. This will not speed up this case:
```
da = xr.DataArray(
        np.arange(6).reshape(3, 2),
        dims=["x", "y"],
        coords={"x": [0, 1, 2], "y": [-0.1, -0.3]},
)

x_new = xr.DataArray([0.5, 1.5, 2.5], dims=["x1"]) 
y_new = xr.DataArray([-0.15, -0.25, -0.35], dims=["x1"])
da.interp(x=x_new, y=y_new)
```

Correct? Is that intentional?

Reply:
Thanks @dcherian 
It was intentional, but you are right. This case can be also improved.

```python
da.interp(x=x_new, y=y_new)
``` 
should be equivalent to
```python
da.interp(x=x_new).interp(y=y_new)
```

But then I'm a bit confused about which case should be improved.
For example if `len(x_new) = len(y_new) = 1000000`, then the original interpretation may be faster, although this is a rare use case.
Maybe we can use some heuristics, such as
```python
len(x_new) < len(x)
```
?

Reply:
I'm now thinking that the simpler behavior is better; for an orthogonal interpolation we interpolate sequentially and otherwise we use `interpn`.
Further improvement may be done in upstream.

Reply:
Sounds good.

Reply:
> ```
> da = xr.DataArray(
>         np.arange(6).reshape(3, 2),
>         dims=["x", "y"],
>         coords={"x": [0, 1, 2], "y": [-0.1, -0.3]},
> )
> 
> x_new = xr.DataArray([0.5, 1.5, 2.5], dims=["x1"]) 
> y_new = xr.DataArray([-0.15, -0.25, -0.35], dims=["x1"])
> da.interp(x=x_new, y=y_new)
> ```

It looks that this case is not slow even with our current code.
The problem is when the final destination is a regular grid, where `interpn` will compute many times.
So, probably this PR should work good enough for this case.